### PR TITLE
tidb: add a new command line flag for redact

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/17156
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: command-line-flags-for-tidb-configuration.md [hunk 1]
```diff
@@ -219,6 +219,12 @@ When you start the TiDB cluster, you can use command-line options or environment
 + Sets the CPU affinity of TiDB servers, which is separated by commas. For example, "1,2,3".
 + Default: `""`
 
+## `--redact`
+
++ Determines whether to enable TiDB server to desensitize log files.
++ Default: false
++ You can use `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the `<input>` log file and output it to `<output>`. When the value is `true`, TiDB desensitizes the log file. When it is `false`, TiDB restores the log file. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```
Comment (hfxsd) [inline command-line-flags-for-tidb-configuration.md:226]:
```suggestion
+ Determines whether TiDB server desensitizes log files when using the subcommand `collect-log`.
+ Default: false
+ When the value is `true`, it is a masking operation, and all fields marked with `‹..›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `< output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```

File: command-line-flags-for-tidb-configuration.md [hunk 2]
```diff
@@ -219,6 +219,12 @@ When you start the TiDB cluster, you can use command-line options or environment
 + Sets the CPU affinity of TiDB servers, which is separated by commas. For example, "1,2,3".
 + Default: `""`
 
+## `--redact`
+
++ Determines whether TiDB server desensitizes log files when using the subcommand `collect-log`.
++ Default: false
++ When the value is `true`, it is a masking operation, and all fields marked with `‹..›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `< output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```
Comment (Defined2014) [inline command-line-flags-for-tidb-configuration.md:226]:
```suggestion
+ When the value is `true`, it is a masking operation, and all fields marked with `‹..›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `<output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```

**Review by Defined2014 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

File: command-line-flags-for-tidb-configuration.md [hunk 3]
```diff
@@ -219,6 +219,12 @@ When you start the TiDB cluster, you can use command-line options or environment
 + Sets the CPU affinity of TiDB servers, which is separated by commas. For example, "1,2,3".
 + Default: `""`
 
+## `--redact`
+
++ Determines whether TiDB server desensitizes log files when using the subcommand `collect-log`.
++ Default: false
++ When the value is `true`, it is a masking operation, and all fields marked with `‹..›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `<output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```
Comment (hfxsd) [inline command-line-flags-for-tidb-configuration.md:226]:
```suggestion
+ When the value is `true`, it is a masking operation, and all fields wrapped in `‹ ›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `<output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```

File: command-line-flags-for-tidb-configuration.md [hunk 4]
```diff
@@ -219,6 +219,12 @@ When you start the TiDB cluster, you can use command-line options or environment
 + Sets the CPU affinity of TiDB servers, which is separated by commas. For example, "1,2,3".
 + Default: `""`
 
+## `--redact`
+
++ Determines whether TiDB server desensitizes log files when using the subcommand `collect-log`.
```
Comment (lilin90) [inline command-line-flags-for-tidb-configuration.md:224]:
```suggestion
+ Determines whether the TiDB server desensitizes log files when using the subcommand `collect-log`.
```

File: command-line-flags-for-tidb-configuration.md [hunk 5]
```diff
@@ -219,6 +219,12 @@ When you start the TiDB cluster, you can use command-line options or environment
...
++ Default: false
++ When the value is `true`, it is a masking operation, and all fields wrapped in `‹ ›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `<output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```
Comment (lilin90) [inline command-line-flags-for-tidb-configuration.md:226]:
```suggestion
+ When the value is `true`, it is a masking operation, and all fields wrapped in `‹ ›` mark symbols are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `<output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**